src/test/java/com/hotels/beeju/HiveServer2JUnitRuleTest.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (C) 2015-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HiveServer2JUnitRuleTest {
+
+  private static final String DATABASE = "my_test_db";
+
+  public @Rule HiveServer2JUnitRule server = new HiveServer2JUnitRule(DATABASE);
+
+  @Before
+  public void init() throws Exception {
+    Class.forName(server.driverClassName());
+  }
+
+  @Test
+  public void databaseNane() {
+    assertThat(server.databaseName(), is(DATABASE));
+  }
+
+  @Test
+  public void showCreateTable() throws Exception {
+    String tableName = "my_table";
+    Table table = createUnpartitionedTable(DATABASE, tableName);
+
+    StringBuilder showCreateTable = new StringBuilder(256);
```

Comment:
Remove the magic 256 number, we don't have to be that efficient in our tests so just use the default constructor.